Tests/CommandsTests/SwiftCommandStateTests.swift

```diff
@@ -533,6 +533,45 @@ final class SwiftCommandStateTests: CommandsTestCase {
             }
         }
     }
+    
+    func testPIDFileHandlerLifecycle() async throws {
+        try withTemporaryDirectory { tmpDir in
+            let scratchPath = tmpDir.appending(component: "scratch")
+            try localFileSystem.createDirectory(scratchPath)
+
+            let pidHandler = PIDFile(scratchDirectory: scratchPath)
+
+            // Ensure no PID file exists initially
+            XCTAssertNil(pidHandler.readPID(), "No PID should exist initially")
+
+            // Write current PID
+            let currentPID = pidHandler.getCurrentPID()
+            try pidHandler.writePID(pid: currentPID)
```

Comment:
suggestion: can we add a test where multiple `writePID(...)` are calls sequentially?